File: iOS/Daily/DesignSystem/ThemeColors.swift
import SwiftUI

public extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3: // RGB (12-bit)
            (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6: // RGB (24-bit)
            (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8: // ARGB (32-bit)
            (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default:
            (a, r, g, b) = (255, 0, 0, 0)
        }
        self.init(
            .sRGB,
            red: Double(r) / 255,
            green: Double(g) / 255,
            blue: Double(b) / 255,
            opacity: Double(a) / 255
        )
    }
}

public struct ThemeColors {
    // WinUI Background stops
    public static let bgStop0 = Color(hex: "030609")
    public static let bgStop1 = Color(hex: "050F1A")
    public static let bgStop2 = Color(hex: "0D1A35")
    public static let bgStop3 = Color(hex: "132B4A")
    
    // Light theme background
    public static let lightBgColor = Color(hex: "D4C9B0")
    
    // Accent Glow & Highlights
    public static let accentCyan = Color(hex: "00E5FF")
    public static let accentBlue = Color(hex: "4A9EFF")
    public static let glowPurple = Color(hex: "8A2BE2")
    
    // Status colors
    public static let error = Color(hex: "FF6B6B")
    public static let warning = Color(hex: "FFD166")
    public static let success = Color(hex: "00E676")
    
    // Foreground
    public static let fgPrimaryDark = Color.white
    public static let fgMutedDark = Color.white.opacity(0.7)
    public static let fgPrimaryLight = Color(hex: "1A1A1A")
    public static let fgMutedLight = Color(hex: "1A1A1A").opacity(0.65)
    
    // Glass styling
    public static let glassDarkBorder = LinearGradient(
        colors: [
            Color.white.opacity(0.35),
            Color.white.opacity(0.10),
            Color.clear,
            Color.white.opacity(0.18)
        ],
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )
    
    public static let glassLightBorder = LinearGradient(
        colors: [
            Color.black.opacity(0.18),
            Color.black.opacity(0.06),
            Color.clear,
            Color.black.opacity(0.12)
        ],
        startPoint: .topLeading,
        endPoint: .bottomTrailing
    )
}

File: Android/core-designsystem/src/main/java/com/intellidream/daily/designsystem/ThemeColors.kt
package com.intellidream.daily.designsystem

import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color

object ThemeColors {
    // Background gradient stops (mirroring iOS ThemeColors & WinUI 3)
    val bgStop0 = Color(0xFF030609)
    val bgStop1 = Color(0xFF050F1A)
    val bgStop2 = Color(0xFF0D1A35)
    val bgStop3 = Color(0xFF132B4A)

    // Light theme background
    val lightBgColor = Color(0xFFD4C9B0)

    // Accent Glow & Highlights
    val accentCyan = Color(0xFF00E5FF)
    val accentBlue = Color(0xFF4A9EFF)
    val accentPink = Color(0xFFFF2D55)
    val accentPurple = Color(0xFFAF52DE)
    val accentGreen = Color(0xFF00E676)
    val accentOrange = Color(0xFFFF9500)
    val glowPurple = Color(0xFF8A2BE2)

    // Status colors
    val error = Color(0xFFFF6B6B)
    val warning = Color(0xFFFFD166)
    val success = Color(0xFF00E676)

    // Foreground / Text
    val fgPrimaryDark = Color.White
    val fgMutedDark = Color.White.copy(alpha = 0.70f)
    val textSecondary = Color.White.copy(alpha = 0.70f)
    val textMuted = Color.White.copy(alpha = 0.45f)
    val fgPrimaryLight = Color(0xFF1A1A1A)
    val fgMutedLight = Color(0xFF1A1A1A).copy(alpha = 0.65f)

    // Glass borders
    val glassDarkBorder = Brush.linearGradient(
        colors = listOf(
            Color.White.copy(alpha = 0.35f),
            Color.White.copy(alpha = 0.10f),
            Color.Transparent,
            Color.White.copy(alpha = 0.18f)
        )
    )

    val glassLightBorder = Brush.linearGradient(
        colors = listOf(
            Color.Black.copy(alpha = 0.18f),
            Color.Black.copy(alpha = 0.06f),
            Color.Transparent,
            Color.Black.copy(alpha = 0.12f)
        )
    )

    val backgroundGradient = Brush.verticalGradient(
        colors = listOf(bgStop0, bgStop1, bgStop2, bgStop3)
    )

    val activeTabIndicatorBrush = Brush.linearGradient(
        colors = listOf(
            accentBlue.copy(alpha = 0.65f),
            accentCyan.copy(alpha = 0.40f)
        )
    )
}
